src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ASN1.cs

```diff
@@ -67,7 +68,7 @@ internal static unsafe string GetOidValue(IntPtr asn1ObjectPtr)
 
             if (bytesNeeded < StackCapacity)
             {
-                return Marshal.PtrToStringUTF8((IntPtr)bufStack, bytesNeeded);
+                return System.Text.Encoding.UTF8.GetString(bufStack, bytesNeeded);
```

Comment:
```suggestion
                return Encoding.UTF8.GetString(bufStack, bytesNeeded);
```
And `using System.Text;` if necessary